### PR TITLE
Vehicle Cooldown

### PR DESCRIPTION
**Pull request by Fate-JH**
The purchase timer for a vehicle is no longer applied on an unfulfilled vehicle purchase.  It only get set once the player is seated in the vehicle and, if they do not get seated through the prescribed method (vehicle spawn pad protocol), it does not get get applied.

The branch is called `cooldown-reset` because originally the purchase timer was still levied at purchase confirmation but, if the order was cancelled, then the cooldown was cancelled too (reset).  This workflow was reverted as no reason requires setting the timer early.  We could use it for communicating that the purchase went through but that confirmation already is performed visually through the automatic closing the terminal interface.

### CODE REVIEW

**Review by jgillich — CHANGES_REQUESTED** [review]

File: src/main/scala/net/psforever/actors/session/SessionActor.scala [hunk 1]
```diff
@@ -2934,6 +2942,12 @@ class SessionActor(middlewareActor: typed.ActorRef[MiddlewareActor.Command], con
         }
         sendResponse(PlanetsideAttributeMessage(vehicle_guid, 22, 1L))          //mount points off
         sendResponse(PlanetsideAttributeMessage(player.GUID, 21, vehicle_guid)) //ownership
+        avatarActor ! AvatarActor.UpdatePurchaseTime(vehicle.Definition)
+        Avatar.purchaseCooldowns.get(vehicle.Definition) match {
+          case Some(cooldown) =>
+            sendResponse(AvatarVehicleTimerMessage(player.GUID, vehicle.Definition.Name, cooldown.toSeconds, unk1 = true))
```
[the remark this answers is not in the document]
Comment (Fate-JH) [reply src/main/scala/net/psforever/actors/session/SessionActor.scala:2948]: I was just moving code from one block to the other.
Looking at that `AvatarActor` code now, do we have to refresh every cooldown each time we update for a specific one?  It feels like that could be better.

[the remark this answers is not in the document]
Comment (jgillich) [reply src/main/scala/net/psforever/actors/session/SessionActor.scala:2948]: It's possible I left that in there. And yes, we could definitely update AvatarActor to only send the timer that changed

[the remark this answers is not in the document]
Comment (Fate-JH) [reply src/main/scala/net/psforever/actors/session/SessionActor.scala]: Only the message is sent now.

File: src/main/scala/net/psforever/actors/session/AvatarActor.scala [hunk 2]
```diff
@@ -1355,4 +1323,48 @@ class AvatarActor(
...
+              )
+
+            case _ =>
+              keysToDrop = keysToDrop :+ key  //key has timed-out
+          }
+        case _ => ;
+      }
+    }
+    if (keysToDrop.nonEmpty) {
+      avatar.copy(purchaseTimes = avatar.purchaseTimes.removedAll(keysToDrop))
```
Comment (jgillich) [inline src/main/scala/net/psforever/actors/session/AvatarActor.scala:1373]:
```scala
avatar = avatar.copy(purchaseTimes = avatar.purchaseTimes.removedAll(keysToDrop))
```

**Review by jgillich — CHANGES_REQUESTED** [review]

**Pull request merged**